core/drivers/crypto/crypto_api/hash/hash.c

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: BSD-2-Clause
```

Comment:
A separate directory for this file seems a bit too much. How about putting this file directly below `crypto_api`?

Reply:
May be but I'll classified all algorithms in a specific directory. For 2 raisons:
* it is easier to find files associated to the algorithms.
* it facilitate the build by including of not the directory.

At the end we will have following directories (at least):
* acipher (ECC/DH/RSA/DSA)
* cipher (AES/DES/DES3)
* hash (HASH)
* mac (HMAC/Cipher MAC/CMAC)